### PR TITLE
ci(check): Add `astro check` step

### PR DESCRIPTION
**Pull request by Princesseuh**
#### What kind of changes does this PR include?

- Something else!

#### Description

This adds `astro check` to the CI, to make sure we don't have type issues in the docs's source code. This PR also fixes some type issues that were preventing it from passing correctly on the current version

### CODE REVIEW

File: src/components/HeadCommon.astro [hunk 1]
```diff
@@ -33,17 +33,16 @@ import '@fontsource/ibm-plex-mono/400-italic.css';
...
+
 		let range = new Range();
-		range.setStart(el.target, 0);
-		range.setEnd(el.target, 1);
-		document.getSelection().removeAllRanges();
-		document.getSelection().addRange(range);
+		range.setStart(node, 0);
+		range.setEnd(node, 1);
+		document.getSelection()?.removeAllRanges();
+		document.getSelection()?.addRange(range);
```
Comment (Princesseuh) [inline src/components/HeadCommon.astro:45]:
```suggestion
		const selection = document.getSelection();
		if (selection) {
			selection.removeAllRanges();
			selection.addRange(range);
		}
```

File: src/components/tabs/TabbedContent.astro [hunk 2]
```diff
@@ -65,15 +65,15 @@ const { tabs } = Astro.props as Props;
 				tab.setAttribute('role', 'tab');
 				tab.setAttribute('id', this.id + 'tab' + this.count++);
 				tab.setAttribute('tabindex', '-1');
-				tab.parentNode.setAttribute('role', 'presentation');
+				(tab.parentNode as HTMLElement)?.setAttribute('role', 'presentation');
```
Comment (Princesseuh) [inline src/components/tabs/TabbedContent.astro:68]:
```suggestion
				tab.parentElement?.setAttribute('role', 'presentation');
```

Comment (delucis) [reply src/components/tabs/TabbedContent.astro:68]: I did not know both of these existed!

**Review by delucis — APPROVED** [review]
Approving from my end! Might be smart to have another pair of docs maintainer eyes have a scroll through the deploy preview in case they catch something I broke? (Risk should be low as most of the changes aren’t functional.)

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks for making `astro check` in CI viable 🙌

**Comment by delucis** [comment]
Yay! Thanks for checking @Yan-Thomas 💜

**Pull request merged**